### ISSUE TITLE
Paste to Typora for aotu-uploading using PicoGo would have PERMISSION ISSUE

### ISSUE THREAD

**Issue by HarborZeng** [typora/typora-issues#3331]
I used to use PicGo in a separate way, but recently found that PicGo is integrated in Typora. So I tried to take a screenshot with WINDOWS 10 截图和草图（*snip and sketch* in English）, and paste it to Typora, but got a error as below.

<img width="250" alt="批注 2020-03-06 141148" src="https://user-images.githubusercontent.com/8568937/76057521-c01b7b80-5fb4-11ea-8835-8191ff969adf.png">

here is my typora settings:

<img width="323" alt="批注 2020-03-06 141921" src="https://user-images.githubusercontent.com/8568937/76057812-8c8d2100-5fb5-11ea-9d96-647ec295bf48.png">

If I run Typora as Administrator, then I would have this in my article:

![image](https://user-images.githubusercontent.com/8568937/76057900-d37b1680-5fb5-11ea-9be1-53fa8a0b3737.png)

I tried to uninstall Typora, downloaded the latest from official website, and reinstalled it, then copy and paste got me this error:

![image](https://user-images.githubusercontent.com/8568937/76059120-02df5280-5fb9-11ea-8ba9-f821a67337b9.png)

What should I do to avoid these **Permission ISSUEs**

FYI: the upload test is passed

**Comment by abnerlee** [comment]
Can you run your custom command correctly in CMD?

**Comment by HarborZeng** [comment]
> Can you run your custom command correctly in CMD?

Thanks for your attention. I am not following about what *custom command* do you mean, because I use *PicGo.app* rather than *PicGo-Core*

In case that you mean this command, 
![image](https://user-images.githubusercontent.com/8568937/76165034-9c139200-618e-11ea-9b11-96b472f5ddf6.png)

I tested in CMD:

![image](https://user-images.githubusercontent.com/8568937/76165041-b0f02580-618e-11ea-836a-ca6afd88541c.png)

It means `using is neither an internal nor an external command or executable app or bash file.`

-----

Here are other work I have done for Typora run as I wish:

## scenario 1

### config 1

![image](https://user-images.githubusercontent.com/8568937/76165119-628f5680-618f-11ea-9533-e86dc74037f5.png)

### config 2

![image](https://user-images.githubusercontent.com/8568937/76165107-56a39480-618f-11ea-9d90-7085a6ab73dc.png)

### test paste

![image](https://user-images.githubusercontent.com/8568937/76165136-881c6000-618f-11ea-9a38-7891bf77515b.png)

### right click and select upload

![image](https://user-images.githubusercontent.com/8568937/76165158-af732d00-618f-11ea-9960-469050ed9e50.png)

It worked.

## scenario 2

### config 1

![image](https://user-images.githubusercontent.com/8568937/76165173-d2054600-618f-11ea-99ba-c07dc5358096.png)

### config 2

![image](https://user-images.githubusercontent.com/8568937/76165185-ec3f2400-618f-11ea-97b4-b97966b4094b.png)

### test paste

something is wrong because nothing happened, so I checked console output:

```console
typora://userData\themes\base.user.css:1 Failed to load resource: net::ERR_FILE_NOT_FOUND
typora://userData\themes\github.user.css:1 Failed to load resource: net::ERR_FILE_NOT_FOUND
fs.js:447 Uncaught (in promise) Error: EPERM: operation not permitted, open 'upload\image-20200308225539084.png'
    at Object.openSync (fs.js:447)
    at Object.func (electron/js2c/asar.js:138)
    at Object.func [as openSync] (electron/js2c/asar.js:138)
    at Object.writeFileSync (fs.js:1206)
    at frame.js:1
```

## scenario 3

run Typora as administrator.

config keeps as scenario 2:

### test paste

![image](https://user-images.githubusercontent.com/8568937/76165248-6c658980-6190-11ea-9c7a-16a55552d149.png)

yes, it pasted, but not auto-uploaded, check the console:

```console
typora://userData\themes\base.user.css:1 Failed to load resource: net::ERR_FILE_NOT_FOUND
typora://userData\themes\github.user.css:1 Failed to load resource: net::ERR_FILE_NOT_FOUND
```

### right click and select open image location

this location is opened `C:\Program Files\Typora\upload` in Windows Explorer. (so I guess this is way an error occurred when not run Typora as Administrator)

### right click and select upload

![image](https://user-images.githubusercontent.com/8568937/76165283-b6e70600-6190-11ea-8096-1f606666572b.png)

It uploaded successfully.

## scenario 4

run as not admin, open already save md file.

configs keep the same as scenario 2

### paste

![image](https://user-images.githubusercontent.com/8568937/76165644-8eacd680-6193-11ea-8605-8b64b37dedaf.png)

## scenario 5

run as admin, open already save md file.

configs keep the same as scenario 2

### paste

![image](https://user-images.githubusercontent.com/8568937/76165674-cfa4eb00-6193-11ea-8178-d773bb3926ca.png)

### output 

```console
typora://userData\themes\base.user.css:1 Failed to load resource: net::ERR_FILE_NOT_FOUND
typora://userData\themes\github.user.css:1 Failed to load resource: net::ERR_FILE_NOT_FOUND
/C:/Users/harbo/Documents/easymark/upload/image-20200308232218961.png?lastModify=1583680937:1 GET file:///C:/Users/harbo/Documents/easymark/upload/image-20200308232218961.png?lastModify=1583680937 net::ERR_FILE_NOT_FOUND
Image (async)
h @ frame.js:1
t.decorate @ frame.js:1
F @ frame.js:1
de.output @ frame.js:1
Te @ frame.js:1
e @ frame.js:1
y @ frame.js:1
P @ frame.js:1
H @ frame.js:1
M @ frame.js:1
(anonymous) @ frame.js:1
doInsertFromURL @ frame.js:1
(anonymous) @ frame.js:1
doInsertFromURLs @ frame.js:1
A @ frame.js:1
(anonymous) @ frame.js:1
Promise.then (async)
H @ frame.js:1
Promise.then (async)
M @ frame.js:1
(anonymous) @ frame.js:1
dispatch @ frame.js:1
g.handle @ frame.js:1
r @ frame.js:1
/C:/Users/harbo/Documents/easymark/upload/image-20200308232218961.png?lastModify=1583680937:1 GET file:///C:/Users/harbo/Documents/easymark/upload/image-20200308232218961.png?lastModify=1583680937 net::ERR_FILE_NOT_FOUND
Image (async)
Node.rawText @ frame.js:1
$.fn.rawText @ frame.js:1
store @ frame.js:1
sync @ frame.js:1
buildDocumentSnapForNode @ frame.js:1
backupDocumentSnap @ frame.js:1
(anonymous) @ frame.js:1
r @ frame.js:1
/C:/Users/harbo/Documents/easymark/upload/image-20200308232218961.png?lastModify=1583680940:1 GET file:///C:/Users/harbo/Documents/easymark/upload/image-20200308232218961.png?lastModify=1583680940 net::ERR_FILE_NOT_FOUND
Image (async)
(anonymous) @ frame.js:1
each @ frame.js:1
each @ frame.js:1
refreshLocalImg @ frame.js:1
window.onWindowFocusForNode @ frame.js:1
(anonymous) @ VM289:1
WebFrame.<computed> @ C:\Program Files\Typora\resources\electron.asar\renderer\api\web-frame.js:64
(anonymous) @ C:\Program Files\Typora\resources\electron.asar\common\api\deprecate.js:114
(anonymous) @ C:\Program Files\Typora\resources\electron.asar\renderer\web-frame-init.js:11
(anonymous) @ C:\Program Files\Typora\resources\electron.asar\renderer\ipc-renderer-internal-utils.js:7
(anonymous) @ C:\Program Files\Typora\resources\electron.asar\renderer\ipc-renderer-internal-utils.js:7
emit @ events.js:200
onMessage @ C:\Program Files\Typora\resources\electron.asar\renderer\init.js:42
/C:/Users/harbo/Documents/easymark/upload/image-20200308232218961.png?lastModify=1583680946:1 GET file:///C:/Users/harbo/Documents/easymark/upload/image-20200308232218961.png?lastModify=1583680946 net::ERR_FILE_NOT_FOUND
Image (async)
(anonymous) @ frame.js:1
each @ frame.js:1
each @ frame.js:1
refreshLocalImg @ frame.js:1
window.onWindowFocusForNode @ frame.js:1
(anonymous) @ VM291:1
WebFrame.<computed> @ C:\Program Files\Typora\resources\electron.asar\renderer\api\web-frame.js:64
(anonymous) @ C:\Program Files\Typora\resources\electron.asar\common\api\deprecate.js:114
(anonymous) @ C:\Program Files\Typora\resources\electron.asar\renderer\web-frame-init.js:11
(anonymous) @ C:\Program Files\Typora\resources\electron.asar\renderer\ipc-renderer-internal-utils.js:7
(anonymous) @ C:\Program Files\Typora\resources\electron.asar\renderer\ipc-renderer-internal-utils.js:7
emit @ events.js:200
onMessage @ C:\Program Files\Typora\resources\electron.asar\renderer\init.js:42
/C:/Users/harbo/Documents/easymark/upload/image-20200308232218961.png?lastModify=1583680965:1 GET file:///C:/Users/harbo/Documents/easymark/upload/image-20200308232218961.png?lastModify=1583680965 net::ERR_FILE_NOT_FOUND
Image (async)
(anonymous) @ frame.js:1
each @ frame.js:1
each @ frame.js:1
refreshLocalImg @ frame.js:1
window.onWindowFocusForNode @ frame.js:1
(anonymous) @ VM292:1
WebFrame.<computed> @ C:\Program Files\Typora\resources\electron.asar\renderer\api\web-frame.js:64
(anonymous) @ C:\Program Files\Typora\resources\electron.asar\common\api\deprecate.js:114
(anonymous) @ C:\Program Files\Typora\resources\electron.asar\renderer\web-frame-init.js:11
(anonymous) @ C:\Program Files\Typora\resources\electron.asar\renderer\ipc-renderer-internal-utils.js:7
(anonymous) @ C:\Program Files\Typora\resources\electron.asar\renderer\ipc-renderer-internal-utils.js:7
emit @ events.js:200
onMessage @ C:\Program Files\Typora\resources\electron.asar\renderer\init.js:42
/C:/Users/harbo/Documents/easymark/upload/image-20200308232218961.png?lastModify=1583680973:1 GET file:///C:/Users/harbo/Documents/easymark/upload/image-20200308232218961.png?lastModify=1583680973 net::ERR_FILE_NOT_FOUND
Image (async)
(anonymous) @ frame.js:1
each @ frame.js:1
each @ frame.js:1
refreshLocalImg @ frame.js:1
window.onWindowFocusForNode @ frame.js:1
(anonymous) @ VM298:1
WebFrame.<computed> @ C:\Program Files\Typora\resources\electron.asar\renderer\api\web-frame.js:64
(anonymous) @ C:\Program Files\Typora\resources\electron.asar\common\api\deprecate.js:114
(anonymous) @ C:\Program Files\Typora\resources\electron.asar\renderer\web-frame-init.js:11
(anonymous) @ C:\Program Files\Typora\resources\electron.asar\renderer\ipc-renderer-internal-utils.js:7
(anonymous) @ C:\Program Files\Typora\resources\electron.asar\renderer\ipc-renderer-internal-utils.js:7
emit @ events.js:200
onMessage @ C:\Program Files\Typora\resources\electron.asar\renderer\init.js:42
/C:/Users/harbo/Documents/easymark/upload/image-20200308232218961.png?lastModify=1583680980:1 GET file:///C:/Users/harbo/Documents/easymark/upload/image-20200308232218961.png?lastModify=1583680980 net::ERR_FILE_NOT_FOUND
Image (async)
(anonymous) @ frame.js:1
each @ frame.js:1
each @ frame.js:1
refreshLocalImg @ frame.js:1
window.onWindowFocusForNode @ frame.js:1
(anonymous) @ VM299:1
WebFrame.<computed> @ C:\Program Files\Typora\resources\electron.asar\renderer\api\web-frame.js:64
(anonymous) @ C:\Program Files\Typora\resources\electron.asar\common\api\deprecate.js:114
(anonymous) @ C:\Program Files\Typora\resources\electron.asar\renderer\web-frame-init.js:11
(anonymous) @ C:\Program Files\Typora\resources\electron.asar\renderer\ipc-renderer-internal-utils.js:7
(anonymous) @ C:\Program Files\Typora\resources\electron.asar\renderer\ipc-renderer-internal-utils.js:7
emit @ events.js:200
onMessage @ C:\Program Files\Typora\resources\electron.asar\renderer\init.js:42
/C:/Users/harbo/Documents/easymark/upload/image-20200308232218961.png?lastModify=1583680997:1 GET file:///C:/Users/harbo/Documents/easymark/upload/image-20200308232218961.png?lastModify=1583680997 net::ERR_FILE_NOT_FOUND
Image (async)
(anonymous) @ frame.js:1
each @ frame.js:1
each @ frame.js:1
refreshLocalImg @ frame.js:1
window.onWindowFocusForNode @ frame.js:1
(anonymous) @ VM300:1
WebFrame.<computed> @ C:\Program Files\Typora\resources\electron.asar\renderer\api\web-frame.js:64
(anonymous) @ C:\Program Files\Typora\resources\electron.asar\common\api\deprecate.js:114
(anonymous) @ C:\Program Files\Typora\resources\electron.asar\renderer\web-frame-init.js:11
(anonymous) @ C:\Program Files\Typora\resources\electron.asar\renderer\ipc-renderer-internal-utils.js:7
(anonymous) @ C:\Program Files\Typora\resources\electron.asar\renderer\ipc-renderer-internal-utils.js:7
emit @ events.js:200
onMessage @ C:\Program Files\Typora\resources\electron.asar\renderer\init.js:42
/C:/Users/harbo/Documents/easymark/upload/image-20200308232218961.png?lastModify=1583681010:1 GET file:///C:/Users/harbo/Documents/easymark/upload/image-20200308232218961.png?lastModify=1583681010 net::ERR_FILE_NOT_FOUND
Image (async)
(anonymous) @ frame.js:1
each @ frame.js:1
each @ frame.js:1
refreshLocalImg @ frame.js:1
window.onWindowFocusForNode @ frame.js:1
(anonymous) @ VM362:1
WebFrame.<computed> @ C:\Program Files\Typora\resources\electron.asar\renderer\api\web-frame.js:64
(anonymous) @ C:\Program Files\Typora\resources\electron.asar\common\api\deprecate.js:114
(anonymous) @ C:\Program Files\Typora\resources\electron.asar\renderer\web-frame-init.js:11
(anonymous) @ C:\Program Files\Typora\resources\electron.asar\renderer\ipc-renderer-internal-utils.js:7
(anonymous) @ C:\Program Files\Typora\resources\electron.asar\renderer\ipc-renderer-internal-utils.js:7
emit @ events.js:200
onMessage @ C:\Program Files\Typora\resources\electron.asar\renderer\init.js:42
/C:/Users/harbo/Documents/easymark/upload/image-20200308232218961.png?lastModify=1583681013:1 GET file:///C:/Users/harbo/Documents/easymark/upload/image-20200308232218961.png?lastModify=1583681013 net::ERR_FILE_NOT_FOUND
Image (async)
(anonymous) @ frame.js:1
each @ frame.js:1
each @ frame.js:1
refreshLocalImg @ frame.js:1
window.onWindowFocusForNode @ frame.js:1
(anonymous) @ VM363:1
WebFrame.<computed> @ C:\Program Files\Typora\resources\electron.asar\renderer\api\web-frame.js:64
(anonymous) @ C:\Program Files\Typora\resources\electron.asar\common\api\deprecate.js:114
(anonymous) @ C:\Program Files\Typora\resources\electron.asar\renderer\web-frame-init.js:11
(anonymous) @ C:\Program Files\Typora\resources\electron.asar\renderer\ipc-renderer-internal-utils.js:7
(anonymous) @ C:\Program Files\Typora\resources\electron.asar\renderer\ipc-renderer-internal-utils.js:7
emit @ events.js:200
onMessage @ C:\Program Files\Typora\resources\electron.asar\renderer\init.js:42
```

**Comment by Wcdaren** [comment]
应该是没权限在 C盘创建一个upload文件夹 你可以使用 管理员模式 启动这个软件  或者  直接在报错那里说的地方 创建一个upload文件夹

**Comment by abnerlee** [comment]
This should be a bug, there's no need to create the "upload" folder.

**Comment by renmu123** [comment]
when l create the "upload" folder,l still cannot use picgo auto-upload function

**Comment by Thunder-z** [comment]
HI！Has this problem been solved in version 0.9.86?

**Comment by HarborZeng** [comment]
> HI！Has this problem been solved in version 0.9.86?

no.

**Comment by abnerlee** [comment]
should be fixed
